asQ/allatonce/form.py

```diff
@@ -59,6 +61,19 @@ def __init__(self,
 
         self.form = self._construct_form()
 
+    def time_update(self, t=None):
+
+        if t is not None:
+            t.assign(t + self.dt*self.aaofunc.ntimesteps)
+            for n in range((self.aaofunc.nlocal_timesteps)):
+                self.time[n].assign(t + self.dt*(self.aaofunc.transform_index(n, from_range='slice', to_range='window') + 1))
+
+        else:
+            self.t0.assign(self.t0 + self.dt*self.aaofunc.ntimesteps)
+            for n in range((self.aaofunc.nlocal_timesteps)):
```

Comment:
The logic for updating the `time` array from `t0` _should_ be the same for `t=None` or `t is not None`, so I think it would be best to take this loop out of the if-block.